Dynatrace/dynatrace-operator
# Generate Dynatrace API clients using swagger

# Description

OpenAPI Dynatrace API schemas are copied to `./third_party/dynatrace_api` directory.
`swagger/generate-rest-client` makefile target creates go clients on the fly and puts them to `./src/dtclient2`

Requirements:
- `docker`
- from _GNU Coreutils_:
	- `md5sum`
	- `tr`
	- `cut`

### ~~Room for improvement~~ DONE :heavy_check_mark: 
_On my machine (Linux, Intel i9-11950H) it takes ~7seconds to generate clients.
It can be avoided by checking API version, and if already generated code matches schemas versions, resign from automatic clients generation._
```
$ time make swagger/generate-rest-client
./hack/build/generate-rest-clients.sh
make swagger/generate-rest-client  
0,07s user 0,08s system 2% cpu 6,551 total
```

## How can this be tested?
At this stage, no regression is a success, so just tests should be passing.
Idempotence of make operations should be preserved, so just run multiple times in a row:
`make go/test`
`make swagger/generate-rest-client`
First time image is downloaded, be patient :).

## Checklist
